### PR TITLE
storage: move the APIObjectVersioner definition to storage

### PR DESCRIPTION
**Pull request by stevekuznetsov**
The means by which we extract and parse the version of an API object is
not specific to etcd3. In order to allow for a generic suite of tests
against any storage.Interface imlpementation, we need this logic to live
outside of the etcd3 package, or import cycles will exist.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/kind cleanup
/sig api-machinery

```release-note
NONE
```

```docs

```

/assign @wojtek-t @liggitt 
Part of https://github.com/kubernetes/kubernetes/issues/109831

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
GoLand's automagic refactor/move tool was no match for our tangled web of imports!

File: test/integration/etcd/etcd_cross_group_test.go [hunk 1]
```diff
@@ -119,11 +119,14 @@ func TestCrossGroupStorage(t *testing.T) {
 				}
 			}
 
-			versioner := etcd3.APIObjectVersioner{}
+			versioner := storage.APIObjectVersioner{}
 			for _, resource := range resources {
 				// clear out the things cleared in etcd
 				versioned := versionedData[resource.Mapping.Resource]
-				versioner.PrepareObjectForStorage(versioned)
+				if err := versioner.PrepareObjectForStorage(versioned); err != nil {
```
Comment (stevekuznetsov) [inline test/integration/etcd/etcd_cross_group_test.go:126]: Everything else was mechanical refactoring, but I noticed this unhandled error and took care of it as well.

Comment (wojtek-t) [reply test/integration/etcd/etcd_cross_group_test.go:126]: great - thanks!

**Pull request merged**